libs/tokens/src/state/tokenLists/tokenListsStateAtom.ts

```diff
@@ -29,10 +29,35 @@ export const listsStatesByChainAtom = atomWithStorage<TokenListsState>('allToken
 export const tokenListsUpdatingAtom = atom<boolean>(false)
 
 export const listsStatesMapAtom = atom((get) => {
-  const { chainId } = get(environmentAtom)
+  const { chainId, widgetAppCode, selectedLists } = get(environmentAtom)
   const allTokenListsInfo = get(listsStatesByChainAtom)
+  const currentNetworkLists = allTokenListsInfo[chainId]
+
+  return Object.keys(currentNetworkLists).reduce<{ [source: string]: ListState }>((acc, source) => {
+    const list = currentNetworkLists[source]
+    const isDefaultList = !list.widgetAppCode
+    const sourceLowerCased = source.toLowerCase()
+
+    // In widget mode
+    if (widgetAppCode) {
```

Comment:
i don't love too much how widget logic get as deep as the "atom" level.

Reply:
I'm also not super happy with it, but this is more appropriate tradeoff I've got.
As you can see, `widgetAppCode ` is a part of `environmentAtom` which is filled in `WidgetTokensListsUpdater`.
So, there are no dependency inversion breach.